File: src/main/java/com/connectflow/model/ActivityLogEntry.java
package com.connectflow.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "activity_log")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ActivityLogEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(columnDefinition = "CHAR(36)")
    private UUID id;

    @Column(name = "user_name")
    private String userName;

    @Column(name = "user_email")
    private String userEmail;

    @Column(name = "action", nullable = false, length = 100)
    private String action;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "http_method", length = 10)
    private String httpMethod;

    @Column(name = "endpoint", length = 500)
    private String endpoint;

    @Column(name = "ip_address", length = 50)
    private String ipAddress;

    /** SUCCESS or FAILURE */
    @Column(name = "status", length = 20)
    private String status;

    @Column(name = "error_message", length = 1000)
    private String errorMessage;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: backend/src/model/ActivityLogEntry.js
'use strict';
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

/**
 * Activity log entry model - matches Java ActivityLogEntry entity and 'activity_log' table.
 */
const ActivityLogEntry = sequelize.define('ActivityLogEntry', {
  id: { 
    type: DataTypes.CHAR(36), 
    primaryKey: true, 
    defaultValue: DataTypes.UUIDV4 
  },
  userName: { 
    type: DataTypes.STRING(255), 
    field: 'user_name' 
  },
  userEmail: { 
    type: DataTypes.STRING(255), 
    field: 'user_email' 
  },
  action: { 
    type: DataTypes.STRING(100), 
    nullable: false 
  },
  description: { 
    type: DataTypes.STRING(500) 
  },
  httpMethod: { 
    type: DataTypes.STRING(10), 
    field: 'http_method' 
  },
  endpoint: { 
    type: DataTypes.STRING(500) 
  },
  ipAddress: { 
    type: DataTypes.STRING(50), 
    field: 'ip_address' 
  },
  status: { 
    type: DataTypes.STRING(20) 
  },
  errorMessage: { 
    type: DataTypes.STRING(1000), 
    field: 'error_message' 
  },
}, {
  tableName: 'activity_log',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: false,
});

module.exports = ActivityLogEntry;
